rainier-plot/src/main/scala/com/stripe/rainier/plot/Jupyter.scala

```diff
@@ -88,6 +88,50 @@ object Jupyter {
         .yLabel(yKey))
   }
 
+  private def shadedRenderer(intervals: Seq[(Double, (Double,Double))]) = new PlotRenderer {
+    def render(plot: Plot, plotExtent: Extent)(implicit theme: Theme): Drawable = {
+
+      val xtransformer = plot.xtransform(plot, plotExtent)
+      val ytransformer = plot.ytransform(plot, plotExtent)
+
+      val points = intervals.map{case (x, (y1, _)) => 
+        Point(xtransformer(x), ytransformer(y1))
+      } ++ intervals.reverse.map{case (x, (_, y2)) =>
+        Point(xtransformer(x), ytransformer(y2))
+      }
+      Polygon(points).filled(HSLA(210, 100, 0, 0.2))
+    }
+  }
+
+  def scatterShaded(seq: Seq[Map[String, Double]], xKey: String, yKey: String)(
+    fn: Double => Seq[Double])(implicit oh: OutputHandler,
+                               extent: Extent): Unit = {
+    val xy = seq.map { s => (s(xKey), s(yKey)) }
+    val scatter = scatterPlot(xy)  
+  
+    val minX = BigDecimal(scatter.xbounds.min)
```

Comment:
Might make sense to move this intervals calculation into `shadedRenderer` since you have the bounds you need in there. I think then `shadedRenderer` would take a function `Double => Seq[Double, (Double, Double)]`